services/brig/src/Brig/Team/API.hs

```diff
@@ -230,69 +251,163 @@ createInvitation' tid mUid inviteeRole mbInviterUid fromEmail body = do
     throwStd emailExists
 
   maxSize <- setMaxTeamSize <$> view settings
-  pending <- lift $ wrapClient $ DB.countInvitations tid
+  pending <- lift $ liftSem $ Store.countInvitations tid
   when (fromIntegral pending >= maxSize) $
     throwStd (errorToWai @'E.TooManyTeamInvitations)
 
   showInvitationUrl <- lift $ liftSem $ GalleyAPIAccess.getExposeInvitationURLsToTeamAdmin tid
 
-  lift $ do
-    iid <- maybe (liftIO DB.mkInvitationId) (pure . Id . toUUID) mUid
-    now <- liftIO =<< view currentTime
-    timeout <- setTeamInvitationTimeout <$> view settings
-    (newInv, code) <-
-      wrapClient $
-        DB.insertInvitation
-          showInvitationUrl
-          iid
-          tid
-          inviteeRole
-          now
-          mbInviterUid
-          email
-          body.inviteeName
-          timeout
-    (newInv, code) <$ sendInvitationMail email tid fromEmail code body.locale
-
-deleteInvitation :: (Member GalleyAPIAccess r) => UserId -> TeamId -> InvitationId -> (Handler r) ()
+  iid <- maybe (liftIO randomId) (pure . Id . toUUID) mUid
+  now <- liftIO =<< view currentTime
+  timeout <- setTeamInvitationTimeout <$> view settings
+  newInv <-
+    lift . liftSem $
+      Store.insertInvitation
+        iid
+        tid
+        inviteeRole
+        now
+        mbInviterUid
+        email
+        body.inviteeName
+        timeout
+  lift $ sendInvitationMail email tid fromEmail newInv.code body.locale
+  inv <- toInvitation showInvitationUrl newInv
+  pure (inv, newInv.code)
+
+deleteInvitation ::
+  (Member GalleyAPIAccess r, Member InvitationCodeStore r) =>
+  UserId ->
+  TeamId ->
+  InvitationId ->
+  (Handler r) ()
 deleteInvitation uid tid iid = do
   ensurePermissions uid tid [AddTeamMember]
-  lift $ wrapClient $ DB.deleteInvitation tid iid
+  lift . liftSem $ Store.deleteInvitation tid iid
 
-listInvitations :: (Member GalleyAPIAccess r) => UserId -> TeamId -> Maybe InvitationId -> Maybe (Range 1 500 Int32) -> (Handler r) Public.InvitationList
-listInvitations uid tid start mSize = do
+listInvitations ::
+  ( Member GalleyAPIAccess r,
+    Member TinyLog r,
+    Member InvitationCodeStore r
+  ) =>
+  UserId ->
+  TeamId ->
+  Maybe InvitationId ->
+  Maybe (Range 1 500 Int32) ->
+  (Handler r) Public.InvitationList
+listInvitations uid tid startingId mSize = do
   ensurePermissions uid tid [AddTeamMember]
   showInvitationUrl <- lift $ liftSem $ GalleyAPIAccess.getExposeInvitationURLsToTeamAdmin tid
-  rs <- lift $ wrapClient $ DB.lookupInvitations showInvitationUrl tid start (fromMaybe (unsafeRange 100) mSize)
-  pure $! Public.InvitationList (DB.resultList rs) (DB.resultHasMore rs)
+  let toInvitations is = mapM (toInvitation showInvitationUrl) is
```

Comment:
inline?

Reply:
Less readable, and it's called in 2 places.